examples/cluster/plot_kmeans_silhouette_analysis.py

```diff
@@ -0,0 +1,123 @@
+"""
+===============================================================================
+Silhouette analysis for sample data clustered using KMeans clustering algorithm
+===============================================================================
+
+Silhouette analysis can be used to study the separation distance between the
+resulting clusters. The silhouette plot displays a measure of how close each
```

Comment:
... and thus provides a way to assess parameters like number of clusters visually.

(Ideally, this example would show that a grid search optimising the mean silhouette value will choose the visually-best option too.)